instance/api/instance.py

```diff
@@ -47,7 +47,9 @@ def provision(self, request, pk=None):
         Start the (re-)provisioning of an instance
         """
         instance = self.get_object()
-        if instance.status not in (instance.EMPTY, instance.READY):
+        if instance.status not in (instance.EMPTY,
+                                   instance.READY,
+                                   instance.ERROR):
```

Comment:
@chromano Sounds good, good idea - and it makes sense that the newly created instance status is at PROGRESS_SUCCESS initially, since it was successfully newly created at that point.

And you're right about the UI issue it would cause to test for progress=running here. This issue runs deeper actually - this would happen when the instance is EMPTY too, which also happens upon creating the instance initially, or when the instance is being reprovisioned, when the previous server is deleted but the new one hasn't been created yet. It feels like it will need a larger refactoring, using purely the server status to give the instance status doesn't work well in these cases.

But I don't want to add this to the scope of your task - we have another task about states coming up, to turn the current messy set_status() method into a proper state machine. This could be addressed as part of that scope. For now, even if there are outstanding issues with it, it feels cleaner to try to organize the code around the notion of progress=running for testing whether we're currently already performing actions on the instance, and thus can't allow the user to launch an action like provisioning.